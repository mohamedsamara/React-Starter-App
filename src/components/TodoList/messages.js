/*
 * TodoList Messages
 *
 * This contains all the text for the TodoList component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  header: {
    id: 'src.components.TodoList.header',
    defaultMessage: 'This is the TodoList component !',
  },
});
