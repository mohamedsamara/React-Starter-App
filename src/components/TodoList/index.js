/**
 *
 * TodoList
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import { withStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';

import messages from './messages';
import styles from './styles.css';

const muiStyles = {
  card: {
    marginBottom: 15,
  },
};

const TodoList = props => {
  const { todos, classes } = props;

  const todoNodes = todos.map(todo => (
    <Card key={todo.id} className={classes.card}>
      <CardActionArea>
        <CardContent>
          <Typography gutterBottom variant="h5" component="h2">
            {todo.title}
          </Typography>
          <Typography component="p">
            Lizards are a widespread group of squamate reptiles, with over 6,000
            species, ranging across all continents except Antarctica
          </Typography>
        </CardContent>
      </CardActionArea>
      <CardActions>
        <Button size="small" color="primary">
          {todo.completed ? 'Completed' : 'Not completed'}
        </Button>
      </CardActions>
    </Card>
  ));

  return <div className={styles.todoList}>{todoNodes}</div>;
};

TodoList.propTypes = {
  todos: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string,
      completed: PropTypes.bool,
      id: PropTypes.number.isRequired,
      userId: PropTypes.number,
    }),
  ),
};

export default withStyles(muiStyles)(TodoList);
