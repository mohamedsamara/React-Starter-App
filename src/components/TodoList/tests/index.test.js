import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import TodoList from '../index';

const todos = [
  {
    userId: 1,
    id: 1,
    title: 'delectus aut autem',
    completed: false,
  },
  {
    userId: 1,
    id: 2,
    title: 'quis ut nam facilis et officia qui',
    completed: false,
  },
];

const renderComponent = (props = {}) =>
  shallow(<TodoList {...props} todos={todos} />);

describe('<TodoList />', () => {
  it('should render an <TodoList> tag', () => {
    const renderedComponent = renderComponent();

    expect(renderedComponent.exists()).toBe(true);
  });

  it('should <TodoList> render correctly', () => {
    const component = renderer.create(<TodoList todos={todos} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have an todos attribute', () => {
    const renderedComponent = renderComponent();
    expect(renderedComponent.prop('todos')).toEqual(todos);
  });
});
