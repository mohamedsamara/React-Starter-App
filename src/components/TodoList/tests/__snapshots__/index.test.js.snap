// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoList /> should <TodoList> render correctly 1`] = `
<div
  className="todoList"
>
  <div
    className="MuiPaper-root-3 MuiPaper-elevation1-6 MuiPaper-rounded-4 MuiCard-root-2 TodoList-card-1"
  >
    <button
      className="MuiButtonBase-root-33 MuiCardActionArea-root-30"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <div
        className="MuiCardContent-root-36"
      >
        <h2
          className="MuiTypography-root-37 MuiTypography-h5-53 MuiTypography-gutterBottom-64"
        >
          delectus aut autem
        </h2>
        <p
          className="MuiTypography-root-37 MuiTypography-body1-46"
        >
          Lizards are a widespread group of squamate reptiles, with over 6,000 species, ranging across all continents except Antarctica
        </p>
      </div>
      <span
        className="MuiCardActionArea-focusHighlight-32"
      />
      <span
        className="MuiTouchRipple-root-102"
      />
    </button>
    <div
      className="MuiCardActions-root-73"
    >
      <button
        className="MuiButtonBase-root-33 MuiButton-root-76 MuiButton-text-78 MuiButton-textPrimary-79 MuiButton-flat-81 MuiButton-flatPrimary-82 MuiButton-sizeSmall-99 MuiCardActions-action-75"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-77"
        >
          Not completed
        </span>
        <span
          className="MuiTouchRipple-root-102"
        />
      </button>
    </div>
  </div>
  <div
    className="MuiPaper-root-3 MuiPaper-elevation1-6 MuiPaper-rounded-4 MuiCard-root-2 TodoList-card-1"
  >
    <button
      className="MuiButtonBase-root-33 MuiCardActionArea-root-30"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <div
        className="MuiCardContent-root-36"
      >
        <h2
          className="MuiTypography-root-37 MuiTypography-h5-53 MuiTypography-gutterBottom-64"
        >
          quis ut nam facilis et officia qui
        </h2>
        <p
          className="MuiTypography-root-37 MuiTypography-body1-46"
        >
          Lizards are a widespread group of squamate reptiles, with over 6,000 species, ranging across all continents except Antarctica
        </p>
      </div>
      <span
        className="MuiCardActionArea-focusHighlight-32"
      />
      <span
        className="MuiTouchRipple-root-102"
      />
    </button>
    <div
      className="MuiCardActions-root-73"
    >
      <button
        className="MuiButtonBase-root-33 MuiButton-root-76 MuiButton-text-78 MuiButton-textPrimary-79 MuiButton-flat-81 MuiButton-flatPrimary-82 MuiButton-sizeSmall-99 MuiCardActions-action-75"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiButton-label-77"
        >
          Not completed
        </span>
        <span
          className="MuiTouchRipple-root-102"
        />
      </button>
    </div>
  </div>
</div>
`;
